NEXTBUILTINS.md

```diff
@@ -0,0 +1,261 @@
+
+The decorators proposal is intentionally minimal: it only includes the kinds of functionality that are popular with TypeScript "experimental" and Babel "legacy" decorators. There is much more that we can do with these building blocks! This document describes some possible future directions.
+
+**Nothing in this document is part of the MVP proposal; they are ideas for possible future proposals.**
+
+## Use-case-driven built-in decorators for classes
+
+A few common patterns for using decorators could have additional built-in support. Some of these would be possible otherwise, but may be more efficient as built-ins; others require additional support which is not exposed through the three basic built-in decorators.
+
+The decorators in this section are not intended to form any sort of orthogonal, minimal base, but rather to solve practical problems. For a more general approach, see the following section.
+
+### `@tracked`
+
+`@tracked` wraps a public or private field with an accessor, so that, after writing to the field, a particular callback is called. As an example:
+
+```js
+class C {
+  @tracked(f) x;
+}
+```
+
+would behave as:
+
+```js
+class C {
+  #x;
+  get x() { return this.#x; }
+  set x(v) { this.#x = v; f.call(this); }
+}
+```
+
+### `@bound`
+
+`@bound` makes a method auto-bound. By being built-in, it can have the following advantages over the `@bound` decorator described in the main README:
+- Works on private methods, not just public methods.
+- Can be applied to the whole class, rather than just a single method, to apply to all methods in the class.
+- Can be implemented more efficiently, e.g., maintaining stable object shape, avoiding runtime metaprogramming commands, potentially tying into the object representation or the way that calls are done to make allocation lazier.
+
+### `@metadata`
+
+The `@metadata` decorator described in the main README could be a built-in decorator, making it so that the metadata can be set up all in one go, rather than incrementally, imperatively produced. For example, this could allow the V8 object boilerplate optimization to be used.
+
+### `@expose`
+
+The `@expose` decorator is used on a private class element to expose access to get and set it. It's basically like `@register`, except that callbacks are passed into the provided function to access the element. For example:
+
+```js
+class C {
+  @expose(f) #x;
+}
+```
+
+would behave as:
+
+```js
+class C {
+  @register(proto => f(proto,
+                       "#x",
+                       function() { return this.#x },
+                       function(value) { this.#x = value }))
+      #x;
+}
+```
+
+`@expose` could be used as a building block for other decorators creating protected-like visibility, access to private elements for debugging or testing, etc.
+
+### `@accessor`
+
+The `@accessor` decorator creates a getter/setter pair which exposes a private field, method or accessor, as mediated by `get` and `set` callbacks. For example:
```

Comment:
I find the name "accessor" strange to apply to *both* get and set; an accessor imo is only "get" and a mutator is only "set".

Reply:
Let's just call them `@getter` and `@setter`; there is no need to have them in the same built-in decorator: it can be otmized the same way.

Reply:
Maybe @reader or @writer, even. I can think of lots of reasons for and against lots of names. PRs or issues welcome to bikeshed further.